ui/src/containers/VolumePage.js

```diff
@@ -0,0 +1,750 @@
+      (pV) => pV.metadata.name === volume.metadata.name,
+    );
+    // find the mapping PVC of this specific volume
+    const volumePVC = pVCList.find(
+      (pVC) => pVC.spec.volumeName === volume.metadata.name,
+    );
+
+    const volumeComputedCondition = computeVolumeCondition(
+      computeVolumeGlobalStatus(volume.metadata.name, volume?.status),
+      volumePV?.status?.phase === STATUS_BOUND
+        ? intl.translate('yes')
+        : intl.translate('no'),
+    );
+
+    let volumeUsedCurren = null;
+    let volumeAlerts = [];
+    let volumeHealth = '';
+
+    // if volume is bounded
+    if (volumePVC) {
+      volumeUsedCurren = volumeUsedCurrentList?.find(
+        (volStat) =>
+          volStat.metric.persistentvolumeclaim === volumePVC.metadata.name,
+      );
+
+      // filter the alerts related to the current volume.
+      volumeAlerts = alerts?.list?.filter(
+        (alert) =>
+          alert.labels.persistentvolumeclaim === volumePVC.metadata.name,
+      );
+    } else {
+      volumeHealth = STATUS_NONE;
+    }
+
+    // THE RULES TO COMPUTE THE HEALTH
+    // compute the volume health based on the severity of the alerts
+    // critical => if there is at least one critical
+    if (volumeAlerts.length) {
+      volumeHealth = volumeAlerts.find(
+        (vol) => vol.labels.severity === STATUS_CRITICAL,
+      )
+        ? STATUS_CRITICAL
+        : STATUS_WARNING;
+    } else if (volumeComputedCondition === ('exclamation' || 'unlink')) {
+      volumeHealth = STATUS_NONE;
+    } else {
+      volumeHealth = STATUS_HEALTH;
+    }
+
+    return {
+      name: volume.metadata.name,
+      node: node.name,
+      usage: volumeUsedCurren?.value[1]
+        ? Math.round(
+            (volumeUsedCurren?.value[1] /
+              (volumePV?.spec?.capacity?.storage &&
+                allSizeUnitsToBytes(volumePV?.spec?.capacity?.storage))) *
+              100,
+          )
+        : 0,
+      status: volumeComputedCondition,
+      bound:
+        volumePV?.status?.phase === STATUS_BOUND
+          ? intl.translate('yes')
+          : intl.translate('no'),
+      storageCapacity:
+        volumePV?.spec?.capacity?.storage || intl.translate('unknown'),
+      storageClass: volume.spec.storageClassName,
+      creationTime: formatDate(volume.metadata.creationTimestamp),
+      usageRawData: volumeUsedCurren?.value[1]
+        ? bytesToSize(volumeUsedCurren?.value[1])
+        : 0,
+      health: volumeHealth,
+      // when the data is not ready, display `loading` for the moment
+      latency: volumeLatencyCurrent
+        ? Math.round(
+            volumeLatencyCurrent?.find(
+              (vLV) => vLV.metric.device === volume.status.deviceName,
+            )?.value[1] * 100,
+          ) + 'ms'
+        : 'loading',
+    };
+  });
+
+  const currentVolume = volumeListData.find(
+    (vol) => vol.name === currentVolumeName,
+  );
+
+  // Define a default UI for filtering
+  function GlobalFilter({
```

Comment:
I believe this should be extract outside of the `VolumePage` component (not necessary in a new file but outside of the component.

Reply:
Extracted the volume table to a separated component.